identifier/src/containers/Login/Login.jsx

```diff
@@ -76,14 +76,29 @@ function Login(props) {
   useEffect(() => {
     if (hello && hello.state && history.action !== 'PUSH') {
       if (!query.prompt || query.prompt.indexOf('select_account') === -1) {
-        dispatch(advanceLogonFlow(true, history));
-        return;
-      }
+          dispatch(advanceLogonFlow(true, history));
+          return;
+        }
 
       history.replace(`/chooseaccount${history.location.search}${history.location.hash}`);
-      return;
-    }
-  }, [ /* no dependencies */ ]); // eslint-disable-line react-hooks/exhaustive-deps
+        return;
+      }
+      if (query && query.login_hint) {
+        // for login hint user name validation
+        let sanitizedHint = query.login_hint;
+        if (
+          !validator.isEmail(query.login_hint) ||
+          !validator.isEmail(`${query.login_hint}@example.com`)
+        ) {
+          sanitizedHint = "";
+        }
+        dispatch(updateInput("username", sanitizedHint));
```

Comment:
This is not the logic I suggested, move the updateInput line into the if condition. No need for a misleading variable.

Reply:
Removed the extra variable all together.